src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NugetApiPackageManager.cs

```diff
@@ -200,7 +202,31 @@ public async Task<(string LatestVersion, bool IsLatestVersion)> GetLatestVersion
             return (package.Identity.Version.ToNormalizedString(), isLatestVersion);
         }
 
-        private async Task<(PackageSource, NugetPackageMetadata)> GetLatestVersionInternalAsync(
+        /// <summary>
+        /// Gets Nuget package metadata.
+        /// </summary>
+        /// <param name="packageIdentifier">NuGet package identifier.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>Nuget package metadata.</returns>
+        /// <exception cref="InvalidNuGetSourceException">when sources passed to install request are not valid NuGet feeds or failed to read default NuGet configuration.</exception>
+        /// <exception cref="PackageNotFoundException">when the package cannot be find in default or source NuGet feeds.</exception>
+        public async Task<PackageMetadata> GetPackageMetadataAsync(string packageIdentifier, CancellationToken cancellationToken = default)
```

Comment:
this method potentially should check the source used to installing the package, not nuget.org